src/components/InternalLinkButton/__tests__/InternalLinkButton.test.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { InternalLinkButton } from '../index';
+
+describe('InternalLinkButton', () => {
+  const component = (
+    <InternalLinkButton
+      size="large"
+      className="split-button__start"
+      href={{
+        pathname: '/[platform]/start/getting-started/introduction/',
+        query: { platform: 'react' }
+      }}
+    >
+      Get started
+    </InternalLinkButton>
+  );
+
+  it('should render the InternalLinkButton component', async () => {
+    render(component);
+    const internalLinkButtonNode = await screen.getByRole('link');
+    expect(internalLinkButtonNode).toBeInTheDocument();
+  });
+
+  it('should link to render correctly', async () => {
+    render(component);
+    const internalLinkButtonNode = await screen.getByRole('link');
+    expect(internalLinkButtonNode.href).toContain(
+      '/react/start/getting-started/introduction'
+    );
+
+    expect(internalLinkButtonNode.textContent).toBe('Get started');
+  });
+
+  it('should render optional attributes', async () => {
+    render(component);
+    const internalLinkButtonNode = await screen.getByRole('link');
+
+    if (component.props.size)
+      expect(internalLinkButtonNode.className).toContain(
+        `amplify-button--${component.props.size}`
+      );
+    if (component.props.variation)
```

Comment:
Curious why you are adding if conditions when you are in control of the rendering of the component. Component above doesn't seem to use variation, do you want to add it for this test?